libs/ui/packages/react/src/pre-ldls/components/VirtualList/VirtualList.tsx

```diff
@@ -1,29 +1,108 @@
-import React, { ReactNode } from "react";
-import { FixedSizeList as List } from "react-window";
-import AutoSizer from "react-virtualized-auto-sizer";
+import React, { useEffect, useRef } from "react";
+import { useVirtualizer } from "@tanstack/react-virtual";
+
+interface VirtualItem {
+  key: string | number | bigint;
+  index: number;
+  start: number;
+  end: number;
+  size: number;
+}
+
+type VirtualListProps = {
+  count: number;
+  itemHeight: number;
+  overscan?: number;
+  LoadingComponent?: React.ReactNode;
+  isLoading?: boolean;
+  hasNextPage?: boolean;
+  scrollToIndex?: number;
+  scrollToTop?: boolean;
+  threshold?: number;
+  fetchNextPage?: () => void;
+  renderItem: (index: number) => React.ReactNode;
+};
 
 export const VirtualList = ({
   count,
   itemHeight,
+  overscan = 5,
+  LoadingComponent,
+  isLoading,
+  hasNextPage = false,
+  scrollToIndex,
+  threshold = 5,
+  fetchNextPage,
   renderItem,
-}: {
-  count: number;
-  itemHeight: number;
-  renderItem: (index: number) => ReactNode;
-}) => {
+}: VirtualListProps) => {
+  const parentRef = useRef<HTMLDivElement>(null);
+
+  const rowVirtualizer = useVirtualizer({
+    count: hasNextPage ? count + 1 : count,
+    getScrollElement: () => parentRef.current,
+    estimateSize: () => itemHeight,
+    overscan: overscan,
+  });
+
+  const scrollToIndexRef = useRef(scrollToIndex);
+  const scrollToTopRef = useRef(false);
+
+  useEffect(() => {
+    if (scrollToIndexRef.current !== undefined) {
```

Comment:
`if (scrollToIndexRef.current)` for consistency with `if (scrollToTopRef.current) {`?